proxyd/consensus_poller.go

```diff
@@ -305,12 +318,14 @@ func (cp *ConsensusPoller) UpdateBackendGroupConsensus(ctx context.Context) {
 				- not banned
 				- with minimum peer count
 				- updated recently
+				- not lagging
 			*/
 			bs := cp.backendState[be]
 			notUpdated := bs.lastUpdate.Add(cp.maxUpdateThreshold).Before(time.Now())
 			isBanned := time.Now().Before(bs.bannedUntil)
 			notEnoughPeers := !be.skipPeerCountCheck && bs.peerCount < cp.minPeerCount
-			if !be.IsHealthy() || be.IsRateLimited() || !be.Online() || notUpdated || isBanned || notEnoughPeers {
+			lagging := (bs.latestBlockNumber < proposedBlock) && uint64(proposedBlock-bs.latestBlockNumber) >= cp.maxBlockLag
```

Comment:
Shouldn't this just be `bs.latestBlockNumber < proposedBlock`? We don't want to route requests to nodes that don't have the proposed block under any circumstances.